src/dc_msg.rs

```diff
@@ -831,98 +832,74 @@ pub unsafe fn dc_msg_get_summarytext(
         return dc_strdup(0 as *const libc::c_char);
     }
 
-    let msgtext_c = (*msg)
-        .text
-        .as_ref()
-        .map(|s| CString::yolo(String::as_str(s)));
-    let msgtext_ptr = msgtext_c.map_or(ptr::null(), |s| s.as_ptr());
-
     dc_msg_get_summarytext_by_raw(
         (*msg).type_0,
-        msgtext_ptr,
+        (*msg).text.as_ref().unwrap(),
         &mut (*msg).param,
         approx_characters,
         (*msg).context,
     )
+    .strdup()
 }
 
-/* the returned value must be free()'d */
+/// get a summary text
 #[allow(non_snake_case)]
-pub unsafe fn dc_msg_get_summarytext_by_raw(
+pub fn dc_msg_get_summarytext_by_raw(
     type_0: Viewtype,
-    text: *const libc::c_char,
+    text: &str,
     param: &mut Params,
     approx_characters: libc::c_int,
     context: &Context,
-) -> *mut libc::c_char {
-    /* get a summary text, result must be free()'d, never returns NULL. */
-    let mut ret;
-    let mut prefix: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut pathNfilename: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut value: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut append_text: libc::c_int = 1i32;
+) -> String {
+    let ret: String;
+    let mut prefix = "".to_string();
+    let mut append_text = true;
     match type_0 {
```

Comment:
I'd not pre-define `prefix` and instead make sure each match arm returns it here: `let prefix = match type_0 {...}`.  I think in general trying to avoid mutable variables is beneficial.

Reply:
I'd do the same with `ret`

Reply:
Aw yes, a lot better